app/routes.py

```diff
@@ -1,2 +1,182 @@
-from flask import Blueprint
+"""
+Something went very, very wrong and I had to scrap my original repository, 
+start over and copy paste my code in. Hence, the lack of commits. 
+I was beginning the deployment stage and accidently deleted the origin remote. 
+This lead me down a very dark path, trying different git commands that I didnt fully understand. 
+It was chaos.
 
+Anyways,
+I didn't have time to do docstrings.
+Also, I experimented with doubling up the route decorators since the functions were 
+pretty much the same for Task & Goals.
+Not sure if this is a no no in real life, or if there is a better way to do it. Let me know!
+"""
+
+
+from flask import Blueprint, jsonify, request, abort, g
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from datetime import datetime
+import requests
+import os
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+
+def slack_bot(text):
+
+    SLACK_KEY = os.environ.get("SLACK_KEY")
+    req_body = {"channel": "task-notifications", "text": text}
+    headers = {"Authorization": f"Bearer {SLACK_KEY}"}
+    path = "https://slack.com/api/chat.postMessage"
+
+    requests.post(path, json=req_body, headers=headers)
+
+
+def validate(model, id):
+
+    try:
+        id = int(id)
+    except ValueError:
+        abort(400)
+    return model.query.get_or_404(id)
+
+
+def sort(model):
+
+    sort = request.args.get("sort")
+    if sort == "asc":
+        model = model.query.order_by(model.title.asc())
+    elif sort == "desc":
+        model = model.query.order_by(model.title.desc())
+
+    return model
+
+
+@goals_bp.before_request
+@tasks_bp.before_request
+def get_model():
+
+    bps = {"tasks": (Task, "task"), "goals": (Goal, "goal")}
+    g.model, g.name = bps[request.blueprint]
+
+
+@goals_bp.route("", methods=["GET"])
+@tasks_bp.route("", methods=["GET"])
+def get_all():
+
+    model = g.model
+    if "sort" in request.args:
+        models = sort(model)
+    else:
+        models = model.query.all()
+
+    return jsonify([model.to_dict() for model in models])
+
+
+@goals_bp.route("/<id>", methods=["GET"])
+@tasks_bp.route("/<id>", methods=["GET"])
+def get_one(id):
+
+    model, name = g.model, g.name
+    model = validate(model, id)
+
+    return {f"{name}": model.to_dict()}
+
+
+@goals_bp.route("", methods=["POST"])
+@tasks_bp.route("", methods=["POST"])
+def create():
+
+    model, name = g.model, g.name
+    request_body = request.get_json()
+
+    try:
```

Comment:
This works, and is very concise! You might consider refactor to check for the specific missing column information to provide these details to the user.